contracts/utils/RetireToucanCarbon.sol

```diff
@@ -0,0 +1,1625 @@
+            IERC20(KLIMA).safeTransfer(msg.sender, tradeDust);
+        }
+        if (_klimaType == sKLIMA) {
+            tradeDust =
+                _amountIn -
+                (_amounts[0] == 0 ? _amounts[1] : _amounts[0]);
+            IERC20(KLIMA).safeIncreaseAllowance(stakingHelper, tradeDust);
+
+            IStakingHelper(stakingHelper).stake(tradeDust);
+
+            IERC20(sKLIMA).safeTransfer(msg.sender, tradeDust);
+        }
+        if (_klimaType == wsKLIMA) {
+            tradeDust =
+                _unwrappedKLIMA -
+                (_amounts[0] == 0 ? _amounts[1] : _amounts[0]);
+            IERC20(KLIMA).safeIncreaseAllowance(stakingHelper, tradeDust);
+
+            IStakingHelper(stakingHelper).stake(tradeDust);
+            IERC20(sKLIMA).safeIncreaseAllowance(wsKLIMA, tradeDust);
+            uint256 wrappedDust = IwsKLIMA(wsKLIMA).wrap(tradeDust);
+            IERC20(wsKLIMA).safeTransfer(msg.sender, wrappedDust);
+        }
+    }
+
+    /**
+        @notice Bonds any accumualted fees to the DAO address if the minimum bond size is met.
+        @param _toucanPool Address of the pool token being retired.
+     */
+    function _bondFees(address _toucanPool) internal returns (bool) {
+        // Bond cumulative fees to the DAO if bond is large enough
+        uint256 poolBalance = IERC20(_toucanPool).balanceOf(address(this));
+
+        uint256 bondPrice = IKlimaBondDepository(bondDepository[_toucanPool])
+            .bondPriceInUSD();
+
+        // Minimum .01 KLIMA bond size
+        if ((poolBalance * (1e9)) / bondPrice > 1e7) {
+            IERC20(_toucanPool).safeIncreaseAllowance(
+                bondDepository[_toucanPool],
+                poolBalance
+            );
+
+            IKlimaBondDepository(bondDepository[_toucanPool]).deposit(
+                poolBalance,
+                bondPrice,
+                DAO
+            );
+        }
+        return true;
+    }
+
+    /**
+        @notice Allow the contract owner to update Klima protocol addresses resulting from possible migrations.
+        @param _selection Int to indicate which address is being updated.
+        @param _newAddress New address for contract needing to be updated.
+        @return bool
+     */
+    function setAddress(uint256 _selection, address _newAddress)
+        external
+        onlyOwner
+        returns (bool)
+    {
+        if (_selection == 0) {
+            KLIMA = _newAddress; // 0; Set new KLIMA address
+        }
+        if (_selection == 1) {
+            sKLIMA = _newAddress; // 1; Set new sKLIMA address
+        }
+        if (_selection == 2) {
+            wsKLIMA = _newAddress; // 2; Set new wsKLIMA address
+        }
+        if (_selection == 3) {
+            staking = _newAddress; // 3; Set new staking address
+        }
+        if (_selection == 4) {
+            stakingHelper = _newAddress; // 4; Set new stakingHelper address
+        }
+
```

Comment:
Make use of `else if`s. Also what happens when `_selection = 5`.. it would return `true`.

I have not used switch case statements in solidity yet. Looks interesting and cleaner.
https://docs.soliditylang.org/en/v0.8.11/yul.html?highlight=switch#switch

Reply:
It's not solidity though. Writing yul/assembly is probably more risky if you dont know what you are doing, but I'd also like to see what that would look like.